x/market/types/errors.go

```diff
@@ -68,4 +71,10 @@ var (
 	ErrOrderMatched = sdkerrors.New(ModuleName, errCodeOrderMatched, "order matched")
 	// ErrOrderClosed order closed
 	ErrOrderClosed = sdkerrors.New(ModuleName, errCodeOrderClosed, "order closed")
+	// ErrOrderExists indicates a new order was proposed overwrite the existing store key
+	ErrOrderExists = sdkerrors.New(ModuleName, errCodeOrderExists, "order already exists in store")
+	// ErrOrderTooEarly to match bid
+	ErrOrderTooEarly = sdkerrors.New(ModuleName, errCodeOrderTooEarly, "order: chain height to low for bidding")
```

Comment:
it's not an error to bid before matching begins.

Reply:
Well this raises the debate on using error types as signals for responses to functionality. However `ValidateCanMatch() error` is using that pattern(from before this change). [Returning an error](https://github.com/ovrclk/akash/blob/bd1c6bf74e89d9227ad8879c27df09d031ef12d5/x/market/handler/endblock.go#L105) to indicate skipping this Order in the potential for OnEndBlock Order bidding loop. 

I agree it would be cleaner to signal skipping the Order with a bool/signal struct rather than error value. Shall we start to shift to that pattern?

Reply:
i see.  it's ok, forget it.